src/filechooser.c

```diff
@@ -471,11 +473,51 @@ handle_open (XdpImplFileChooser *object,
           GtkFileFilter *filter;
 
           filter = gtk_file_filter_new_from_gvariant (variant);
+          filters = g_slist_append (filters, g_object_ref (filter));
           gtk_file_chooser_add_filter (GTK_FILE_CHOOSER (dialog), filter);
           g_variant_unref (variant);
         }
       g_variant_iter_free (iter);
     }
+
+  if (g_variant_lookup (arg_options, "current_filter", "@(sa(us))", &current_filter))
+    {
+      g_autoptr (GtkFileFilter) filter = NULL;
+      const char *current_filter_name;
+
+      filter = gtk_file_filter_new_from_gvariant (current_filter);
+      current_filter_name = gtk_file_filter_get_name (filter);
+
+      if (!filters)
+        {
+          /* We are setting a single, unchangeable filter. */
+          gtk_file_chooser_set_filter (GTK_FILE_CHOOSER (dialog), filter);
+        }
+      else
+        {
+          /* We are trying to select the default filter from the list of
+           * filters. We want to naively take filter and pass it to
+           * gtk_file_chooser_set_filter(), but it's not good enough
+           * because GTK+ just compares filters by pointer value, so the
+           * pointer itself has to match. We'll use the heuristic that
+           * if two filters have the same name, they must be the same
+           * unless the application is very dumb.
+           */
+          for (GSList *l = filters; l; l = l->next)
+            {
+              GtkFileFilter *f = l->data;
+              const char *name = gtk_file_filter_get_name (f);
+
+              if (g_strcmp0 (name, current_filter_name) == 0)
+                {
+                  gtk_file_chooser_set_filter (GTK_FILE_CHOOSER (dialog), f);
+                  break;
+                }
+            }
+        }
```

Comment:
If we don't find the current_filter in the list, this silently does nothing. Might be nice to at least warn if we ignore the current_filter value